x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -204,7 +209,7 @@ export async function handleInstallPackageFailure({
   spaceId: string;
   authorizationHeader?: HTTPAuthorizationHeader | null;
 }) {
-  if (error instanceof FleetError) {
+  if (error instanceof ConcurrentInstallOperationError) {
```

Comment:
Yeah I think this is the right call - there's been quite a few SDH's recently with this exact case around "stuck" packages. We should transition the installation state to `failed` unless there's another concurrent installation pending.

Reply:
Good catch, I agree we shouldn't leave a package in installing if the installation failed.

I'm not sure that `ConcurrentInstallOperationError` can only happen when there is an actual concurrent installation. I think I saw this error even when there were no other install requests running.
It would be great to test these failure/concurrent scenarios (even manually) to make sure the installing state is temporary.

Reply:
Yes looks like we throw that `ConcurentInstallOperationError` [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/epm/packages/_install_package.ts#L345-L353) too and not sure we should.